consensus/src/consensus_provider.rs

```diff
@@ -45,11 +50,45 @@ pub fn start_consensus(
         node_config.consensus.mempool_txn_pull_timeout_ms,
         node_config.consensus.mempool_executed_txn_timeout_ms,
     ));
+
     let execution_correctness_manager = ExecutionCorrectnessManager::new(node_config);
-    let state_computer = Arc::new(ExecutionProxy::new(
-        execution_correctness_manager.client(),
-        state_sync_client,
-    ));
+
+    let state_computer: Arc<dyn StateComputer> = if node_config.consensus.decoupled {
+        let guage_e = IntGauge::new("D_EXE_CHANNEL_COUNTER", "counter for the decoupling execution channel").unwrap();
+        let guage_c = IntGauge::new("D_COM_CHANNEL_COUNTER", "counter for the decoupling committing channel").unwrap();
+        let (sender_exec, receiver_exec) = channel::new::<(Vec<Block>, LedgerInfoWithSignatures)>(
+            node_config.consensus.channel_size,
+            &guage_e
+        );
+        let (sender_comm, receiver_comm) = channel::new::<(Vec<ExecutedBlock>, LedgerInfoWithSignatures)>(
+            node_config.consensus.channel_size,
+            &guage_c
+        );
+
+        let execution_phase = ExecutionPhase::new(
+            receiver_exec,
+            execution_correctness_manager.client(),
+            sender_comm,
+        );
+        let commit_phase = CommitPhase::new(
+            receiver_comm,
+            execution_correctness_manager.client(),
+            state_sync_client,
```

Comment:
we can reuse the compute/commit from ExecutionProxy to keep the implementation consistent